### PR TITLE
refactor(bazel): Prepare for non-GitHub url parsing

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Refactor archive suffix stripping
- Create url parsing function that dispatches its execution based on the value of host

## Context

- Ref: #7945 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/bazel/rules/http.ts [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
-  urlString: string | undefined | null
-): UrlParsedResult | null {
-  if (!urlString) {
-    return null;
-  }
-  const url = parseUrl(urlString);
-  if (!url || url.host !== 'github.com' || !url.pathname) {
-    return null;
-  }
+const archives = [
```
Comment (JamieMagee) [inline lib/modules/manager/bazel/rules/http.ts:10]: What's the source for this list? I thought Bazel supported `.tar.zst` as well?

Comment (zharinov) [reply lib/modules/manager/bazel/rules/http.ts:10]: It's from my head, I just needed to refactor it.
Now I wonder which other cool extensions we now have in 2023.

Comment (JamieMagee) [reply lib/modules/manager/bazel/rules/http.ts:10]: Is this the source? https://bazel.build/rules/lib/repo/http

Comment (zharinov) [reply lib/modules/manager/bazel/rules/http.ts:10]: Hah, thanks!

**Review by JamieMagee — DISMISSED** [review]

**Review by JamieMagee — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]

**Review by viceice — APPROVED** [review]
code looks good 🤷‍♂️

**Pull request merged**